python/pyspark/broadcast.py

```diff
@@ -82,7 +82,7 @@ def dump(self, value, f):
         except pickle.PickleError:
             raise
         except Exception as e:
-            msg = "Could not serialize broadcast: " + e.__class__.__name__ + ": " + e.message
+            msg = "Could not serialize broadcast: " + e.__class__.__name__ + ": " + str(e)
```

Comment:
Good call, I forgot about Python 2's `str`/`unicode` behavior.  It could be a problem if an exception includes user input as part of the message.  IMO it's worth handling it rigorously since throwing in the `except:` hides the original error.  I'll look at fixes tonight, in the worse case we could add a `get_exception_message()` helper.